### PR TITLE
MGMT-7389: Create cluster with a platform

### PR DESCRIPTION
**Pull request by michaellevy101**
# Assisted Pull Request

## Description
Currently for creating a vsphere cluster we have to call 2 APIs:
registerCluster and then updateCluster to set the platform
Now we can create a vsphere cluster with a single API call
Relevant for the test-infra and where the user comes from the UI vSphere section rather than Bare Metal.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by eliorerz** [comment]
Please add description and unit-tests

**Comment by gamli75** [comment]
@avishayt why do we need it in the create cluster?

**Comment by avishayt** [comment]
If I know I want vSphere, why do I need 2 API calls?  For example, k8s API, or in the cloud UI if the user comes from vSphere section rather than Bare Metal.

**Comment by gamli75** [comment]
@michaellevy101 please retitle your PR to: MGMT-7389**:** Create cluster with a platform

**Comment by michaellevy101** [comment]
@avishayt you need to create the cluster with the vsphere platform...

File: swagger.yaml [hunk 1]
```diff
@@ -4184,6 +4184,10 @@ definitions:
       network_configuration:
         $ref: '#/definitions/network_configuration'
         x-nullable: true
+      platform:
+        type: object
+        $ref: '#/definitions/platform'
+        x-nullable: true
```
Comment (avishayt) [inline swagger.yaml:4190]: If you do this, can you remove the default logic in the code?

```suggestion
        default: 'baremetal'
```

Comment (michaellevy101) [reply swagger.yaml:4190]: What about kube-api which calls internally?

Comment (avishayt) [reply swagger.yaml:4190]: Good point.  @filanov what's our convention?

Comment (filanov) [reply swagger.yaml:4190]: As far as i remember default doesn't really work. fist need to check that.
In kube-api we use internal functions so the default (if it works) is done by the infra wrappers that we don't call. 
So if we use this feature we will have to handle it in kube-api. 
There are two options There:
* If it's part of the API (AgentClusterInstall) then we will can add default with webhooks (wip).
* If it's not part of the API we will need to handle it in the code of the controller when calling create.

Comment (michaellevy101) [reply swagger.yaml:4190]: I think that instead of dealing with it twice we can set it on the registration operation (All the method has default values)

Comment (avishayt) [reply swagger.yaml:4190]: I agree, the way it's implemented here looks best.

Comment (filanov) [reply swagger.yaml:4190]: +1

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Comment by filanov** [comment]
@michaellevy101 just making sure but are existing clusters covered in this RP?
What will happen to clusters that were created before?

**Pull request merged**